pkg/buildinfo/buildinfo.go

```diff
@@ -15,38 +16,58 @@ import (
 
 // Version identifies the version of Elvish. On development commits, it
 // identifies the next release.
-const Version = "v0.16.0"
+const Version = "0.16.0"
 
-// VersionSuffix is appended to Version in the output of "elvish -version" and
-// "elvish -buildinfo" to build the full version string. This can be overriden
-// when building Elvish; see PACKAGING.md for details.
+// VersionSuffix is appended to Version to build the full version string. It is public so it can be
+// overridden when building Elvish; see PACKAGING.md for details.
 var VersionSuffix = "-dev.unknown"
 
 // Reproducible identifies whether the build is reproducible. This can be
-// overriden when building Elvish; see PACKAGING.md for details.
+// overridden when building Elvish; see PACKAGING.md for details.
 var Reproducible = "false"
 
 // Program is the buildinfo subprogram.
 var Program prog.Program = program{}
 
+type Buildinfo struct {
+	Version      string `json:"version"`
+	Reproducible bool   `json:"reproducible"`
+	GoVersion    string `json:"goversion"`
+}
+
+func (Buildinfo) IsStructMap() {}
+
 type program struct{}
 
 func (program) ShouldRun(f *prog.Flags) bool { return f.Version || f.BuildInfo }
 
-func (program) Run(fds [3]*os.File, f *prog.Flags, _ []string) error {
-	fullVersion := Version + VersionSuffix
-	if f.Version {
-		fmt.Fprintln(fds[1], fullVersion)
-		return nil
+func GetBuildInfo() Buildinfo {
```

Comment:
This needs to have a godoc, and it can be a simple `var` declaration instead of a function.

(If you're worried that the overridden values won't be picked up by a `var` declaration, those overrides happen at link time and directly modify the initial value of the variables.)

Reply:
Good point. This is a function because an earlier iteration of this change had more complicated logic in a misguided attempt to normalize the components of the full version string.